java-lib/src/main/java/com/wavefront/ingester/StringParser.java

```diff
@@ -0,0 +1,113 @@
+package com.wavefront.ingester;
+
+/**
+ * A lightweight parser custom-tailored to suit all of our supported line protocols.
+ *
+ * @author vasily@wavefront.com
+ */
+public class StringParser {
+  private int currentIndex = 0;
+  private final String input;
+  private String peek = null;
+
+  /**
+   * @param input string to parse at instance creation
+   */
+  public StringParser(String input) {
+    this.input = input;
+    this.currentIndex = 0;
+  }
+
+  /**
+   * Retrieves the next available token, but does not advance the further, so multiple
+   * calls to peek() return the same value. The value is cached so performance
+   * penalty is negligible.
+   *
+   * @return next available token or null if end of line was reached
+   */
+  public String peek() {
+    if (input == null) throw new IllegalStateException("parse() must be called first");
```

Comment:
I don't see a parse() call

Reply:
good catch, thanks! fixed. 
it was a carry-over from the first take where parser was threadlocal and re-usable. as it turned out, in jdk11 threadlocal carries about 9-10% performance penalty compared to just creating a new instance every time (creating a new parser is very cheap), so it didn't make much sense to keep it reusable.